packages/medusa/src/loaders/index.ts

```diff
@@ -153,21 +153,21 @@ export default async ({
 
   const plugins = getResolvedPlugins(rootDirectory, configModule, true) || []
   const pluginLinks = await resolvePluginsLinks(plugins, container)
-  await registerWorkflows(plugins)
 
   const { onApplicationShutdown, onApplicationPrepareShutdown } =
     await loadMedusaApp({
       container,
       linkModules: pluginLinks,
     })
 
+  await registerWorkflows(plugins)
```

Comment:
thought: The workflow engine will attempt to retry failed workflows when the module is booting, so this _might_ cause issues because the workflows are not yet registered. Maybe we can test this?

https://github.com/medusajs/medusa/blob/1d40b3cc98faaaf5c2b5ff9601a16f01d970d95e/packages/modules/workflow-engine-redis/src/utils/workflow-orchestrator-storage.ts#L54-L79

Reply:
Logically speaking, the modules have to be ready before any workflows are registered and run, so I think it makes sense to register the workflows after the app is loaded. But what you say can probably happen, so let me test it and see if we can ensure workflows are retried only after the app has fully loaded.

Reply:
To prevent this, maybe the workflow engine can start listening to its queue `onApplicationStart`
That will prevent from start picking jobs before the app is fully loaded

Reply:
+1 – that sounds like a more appropriate approach 👍

Reply:
@carlos-r-l-rodrigues @olivermrbl I was doing exactly that, see the second commit 😄